components/count-badge/count-badge.js

```diff
@@ -48,14 +50,45 @@ class CountBadge extends RtlMixin(LitElement) {
 			 */
 			hideZero: {
 				type: Boolean,
-				reflect: true,
 				attribute: 'hide-zero'
 			},
+			/**
+			 * The text that will show as an aria-label on the badge. NOTE: Only the text will be read by screen-readers (not the number).
```

Comment:
Since this is now the tooltip text as well, maybe something like:
```suggestion
			 * Descriptive text for the badge which will act as an accessible label and tooltip text when tooltips are enabled. Only the text will be read by screen-readers (not the number), so include the number in the text.
```

The bit about how screen readers don't read the number could maybe even just move down to the "Accessibility Properties" section in the README.